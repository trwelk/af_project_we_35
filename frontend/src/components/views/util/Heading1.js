import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import CircularProgress from '@material-ui/core/CircularProgress';
// Styling the component
const useStyles = makeStyles({
    root: {
        maxWidth: 345,
        background: "#80808017"
    },
    headh2: {
        color: "rgba(0, 0, 0, 0.87)",
        fontSize: "36px",
        fontFamily: 'Roboto Condensed',
        fontWeight: 700,
        lineHeight: 1.235,
        textTransform: "uppercase",
    },
    headSpan: {
        width: "55px",
        height: "4px",
        margin: "8px auto 0",
        display: "block",
        backgroundColor: " #ff3366"
    },
    head1container: {
        display: 'flex',
        flexDirection: 'column',
        alignItems:'center'
    }
});

export default function Heading1(props) {
    const classes = useStyles();
    const data = props.data

    //************************Rendering UI Elements*************************************** */

    /*Check if news has been loaded ? Then render the Card component : else render a loading spinner */
    const dataDisplayElement = data ? (
        <div className={classes.head1container}>
            <h2 className={classes.headh2}>
                {data.heading}

            </h2>
            <span className={classes.headSpan}>

            </span>
        </div>
    ) : (
            <CircularProgress />
        )

    return (
        <div >
            {dataDisplayElement}
        </div>
    );
}
