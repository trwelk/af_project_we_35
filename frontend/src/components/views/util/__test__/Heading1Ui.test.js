import React from 'react';
import renderer from 'react-test-renderer';
import Heading1 from '../Heading1';


it('renders Heading 1 correctly', () => {
  const tree = renderer
    .create(<Heading1 data={{heading:"heading"}}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});