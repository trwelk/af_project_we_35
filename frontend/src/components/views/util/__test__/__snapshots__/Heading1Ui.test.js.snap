// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Heading 1 correctly 1`] = `
<div>
  <div
    className="makeStyles-head1container-4"
  >
    <h2
      className="makeStyles-headh2-2"
    >
      heading
    </h2>
    <span
      className="makeStyles-headSpan-3"
    />
  </div>
</div>
`;
